src/Xml/XmlParser.cpp

```diff
@@ -570,7 +570,7 @@ void XmlDocument::streamIn(std::istream& in, std::string& tag)
 			if (node)
 			{
 				node->streamIn(in, tag);
-				bool isElement = node->toElement() != 0;
+				bool isElement = node->toElement() != nullptr;
```

Comment:
Ditto

Reply:
This one seems slightly odd to remove, as that leaves an assignment operator with an implicit type conversion. I suppose it's not much different than the others, though it kind of feels different here.

Reply:
Actually, come to think of it, [ES.87](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es87-dont-add-redundant--or--to-conditions) is worded to apply specifically to conditions. This is not a conditional per say, but rather an assignment expression. I'm not sure that ES.87 really applies in spirit here.

The other thought I had, was for custom types, such code would fail if the type conversion operator was made `explicit`. I suppose we're using the built in type conversion operator for pointers though, which is not `explicit`.

I'm kind of on the fence about this one.

Reply:
Yeah, I'm going to agree with you on this one. It's okay to leave it.